File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/Image/ImageCreateDto.cs
namespace eRezervisi.Common.Dtos.Image
{
    public class ImageCreateDto
    {
        public string ImageBase64 { get; set; } = null!;
        public string ImageFileName { get; set; } = null!;
        public bool IsThumbnail { get; set; }
    }
}

File: erezervisi_desktop/erezervisi_desktop/lib/models/requests/image/image_create_dto.dart
import 'package:image_picker/image_picker.dart';

class ImageCreateDto {
  num? id;
  String? imageBase64;
  String? imageFileName;
  XFile? image;
  String? imageUrl;
  late bool isThumbnail;
  bool delete;

  ImageCreateDto(
      {this.id,
      this.imageBase64,
      this.imageFileName,
      required this.isThumbnail,
      this.imageUrl,
      this.image,
      this.delete = false});

  Map<String, dynamic> toJson() {
    return {
      'id':id,
      'imageBase64': imageBase64,
      'imageFileName': imageFileName,
      'isThumbnail': isThumbnail,
      'delete': delete
    };
  }
}
